### PR TITLE
IOS7: Save state to a separate file when closing ScummVM

### PR DESCRIPTION
**Pull request by criezy**
The state information was saved using ConfMan in the scummvm ini file, but this could cause corruption of that ini file if the application was killed while ConfMan was flushed (see  [bug #14623](https://bugs.scummvm.org/ticket/14623)). To reduce the risk of losing the config file, we now save the state to a separate file.

This does not fix the root cause of [bug #14623](https://bugs.scummvm.org/ticket/14623), but removes one of the main context in which it could occur.

### CODE REVIEW

File: backends/platform/ios7/ios7_osys_main.cpp [hunk 1]
```diff
@@ -280,49 +282,64 @@ void OSystem_iOS7::saveState() {
...
 		int saveSlot = g_engine->getAutosaveSlot();
+		if (saveSlot == -1)
+			return;
 		// Make sure we do not overwrite a user save
 		SaveStateDescriptor desc = g_engine->getMetaEngine()->querySaveMetaInfos(targetName.c_str(), saveSlot);
 		if (desc.getSaveSlot() != -1 && !desc.isAutosave())
 			return;
 
-		// Do the auto-save, and if successful store this it in the config
+		// Do the auto-save, and if successful store this it in the uset settings
```
Comment (larsamannen) [inline backends/platform/ios7/ios7_osys_main.cpp:292]: "uset" should be user I guess

Comment (criezy) [reply backends/platform/ios7/ios7_osys_main.cpp:292]: Right. But that edit was actually done in a first pass in which I considered using `NSUserDefaults`, and I ended up not doing that. So `user settings` isn't accurate either. I will change that.

**Comment by criezy** [comment]
> The most severe thing that could happen now is that the user looses the autosave right? But the config ini won't be wiped thanks to this.

There is indeed still a risk that the autosave itself might be corrupted (and thus losing a previous autosave if there was one) if scummvm is killed while the engine is creating that save. This is one of the reason why this PR is a quick fix to mitigate the issue, but not a proper fix for the bug.

And there is also a risk that the state file might be incomplete, but in that case it is just ignored, so it is the same as if we didn't even attempt to save the state.

Also note to self: when saving the state we may want to check that the engine also supports `kSupportsLoadingDuringStartup`. Otherwise the state cannot be restored, so there is no point saving it (although the autosave could still be useful I guess, even if it has to be reloaded manually?).

**Comment by larsamannen** [comment]
LGTM. I will give it a spin later.

**Pull request merged**